src/object/object_primitive.c

```diff
@@ -15700,3 +15961,94 @@ mr_cmpval_enumeration (DB_VALUE * value1, DB_VALUE * value2, int do_coercion, in
 
   return MR_CMP (s1, s2);
 }
+
+/*
+ * do_get_compressed_data_from_buffer ()	  - Uncompresses a data stored in buffer.
+ *
+ * return()					  : NO_ERROR or error code.
+ * buf(in)					  : The buffer from which is needed decompression.
+ * data(out)					  : The result of the decompression. !!! Needs to be alloc'ed !!!
+ * compressed_size(in)				  : The compressed data size.
+ * uncompressed_size(in)			  : The uncompressed data size.
+ */
+int
+do_get_compressed_data_from_buffer (OR_BUF * buf, char *data, int compressed_size, int uncompressed_size)
+{
+  int rc = NO_ERROR;
+
+  /* Check if the string needs decompression */
+  if (compressed_size > 0)
+    {
+      lzo_uint unc_size = 0;
+      /* Handle decompression */
+
+      /* decompressing the string */
+      rc = lzo1x_decompress ((lzo_bytep) buf->ptr, (lzo_uint) compressed_size, data, &unc_size, NULL);
+      if (rc != LZO_E_OK)
+	{
+	  return rc;
+	}
+      if (unc_size != uncompressed_size)
+	{
+	  /* Decompression failed. It shouldn't. */
+	  assert (false);
+	}
+    }
+  else
+    {
+      /* String is not compressed and buf->ptr is pointing towards an array of char's of length equal to
+       * uncompressed_size */
+
+      rc = or_get_data (buf, data, uncompressed_size);
+    }
+
+  return rc;
+}
+
+/* 
+ * or_get_compression_length()		  - Simulate a compression to find its length to be stored on the disk.
+ * 
+ * return()				  : The length of the compression, based on the new encoding of varchar.
+ * string(in)				  : The string to be compressed.
+ * charlen(in)				  : The length of the string to be compressed.
+ */
+int
+or_get_compression_length (char *string, int charlen)
+{
+  lzo_voidp wrkmem;
```

Comment:
Please investigate if possible to keep a wrkmem object in thread_entry (or global, if on CAS).
(We may need to clear/reset such cached object before or after each LZO compress/decompress).
This should avoid frequent allocation of memory for such objects.